### PR TITLE
Migrated Meshmates Page

### LINKED ISSUE #850: [NG] Migrate the Meshmates page to Layer5-ng
**Current Behavior**
The current website has [Meshmates](https://layer5.io/community/meshmates) page under the community section.

**Desired Situation**
Migrate the Meshmates page under the [community folder](https://github.com/layer5io/layer5/tree/layer5-ng/src/pages/community) in pages section.

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by geraldmaboshe**
Signed-off-by: Gerald Maboshe <maboshegerald1@gmail.com>

**Description**
Migrated meshmates page

This PR fixes #850 

**Notes for Reviewers**
N/A

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by geraldmaboshe** [comment]
@Jashpatel1

**Comment by Jashpatel1** [comment]
> @Jashpatel1

You will have to equate the values with "yes" instead of "true".

**Comment by geraldmaboshe** [comment]
> > @Jashpatel1
> 
> You will have to equate the values with "yes" instead of "true".

Cool, thanks

File: src/sections/Meshmates-grid/ProgramDetails.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+export default function ProgramDetails() {
+    return (
+        <div>
+            <h3>Engaging with a MeshMate</h3>
+            <img
+                src={Program}
+                width="40%" align="right" style={{ margin: "5px" }}
+            />
+
+            <h6>What to Expect</h6>
```
Comment (Jashpatel1) [inline src/sections/Meshmates-grid/ProgramDetails.js:13]: Can you increase this heading size?

File: src/sections/Meshmates-grid/ProgramDetails.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                tips on how to have the best community experience possible.
+  </p>
+            <p>
+                Meshtees are encouraged to get to know their MeshMate as soon as they are
+                assigned (MeshMates and Mentees will be introduced in the Layer5 Slack).
+                Help your MeshMate understand your current skills, ideal topics of learning,
+                and areas of passion. Doing so will help them to point out various aspects
+                of projects that you might find your first foothold.
+  </p>
+            <h6>Meeting Your MeshMate</h6>
```
Comment (Jashpatel1) [inline src/sections/Meshmates-grid/ProgramDetails.js:30]: Same here.

File: src/sections/Meshmates-grid/ProgramDetails.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                        ><a
+                            href="https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+archived%3Afalse+org%3Alayer5io+label%3A%22help+wanted%22+"
+                        >Contribute</a
+                            ></b
+                        >
+        - Grab an open issue or suggest a new one.
+      </p>
+                </li>
+            </ul>
+            <h6>Code of Conduct</h6>
```
Comment (Jashpatel1) [inline src/sections/Meshmates-grid/ProgramDetails.js:91]: Same here

**Review by Jashpatel1 — COMMENTED** [review]
Rest looks fine. Thanks @geraldmaboshe

File: src/sections/Meshmates-grid/ProgramDetails.js (hunk 2 above)
Comment (geraldmaboshe) [reply src/sections/Meshmates-grid/ProgramDetails.js:30]: To h3?

Comment (Jashpatel1) [reply src/sections/Meshmates-grid/ProgramDetails.js:30]: h4 would work I guess.

File: src/pages/community/meshmates.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React, { Component } from "react";
```
Comment (Tanuj22) [inline src/pages/community/meshmates.js:1]: `Component` can be removed

File: src/sections/Emeritus-grid/Emeritus.js [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import React from "react";
+import { useStaticQuery, graphql } from "gatsby";
+import EmeritusGrid from './index'
+
+const Emeritus=props => {
+    const data=useStaticQuery(
```
Comment (Tanuj22) [inline src/sections/Emeritus-grid/Emeritus.js:6]: This query can directly be used in the `EmeritusGrid` file. No need for a separate file.

File: src/sections/Meshmates-grid/Meshmates.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                    filter: { fields: { collection: { eq: "members" } }, frontmatter: { meshmate: { eq: "yes" } } }
+                ) {
+                    nodes {
+                        id
+                        frontmatter {
+                            name
+                            github
+                            twitter
+                            status
+                            meshmate
```
Comment (Tanuj22) [inline src/sections/Meshmates-grid/Meshmates.js:20]: meshmate can be removed as it points to this page itself

File: src/sections/Emeritus-grid/Emeritus.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                    filter: { fields: { collection: { eq: "members" } }, frontmatter: { emeritus: { eq: "yes" } } }
+                ) {
+                    nodes {
+                        id
+                        frontmatter {
+                            name
+                            github
+                            twitter
+                            status
+                            meshmate
```
Comment (Tanuj22) [inline src/sections/Emeritus-grid/Emeritus.js:20]: can be removed

File: src/sections/Emeritus-grid/Emeritus.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+                            emeritus
```
Comment (Tanuj22) [inline src/sections/Emeritus-grid/Emeritus.js:21]: can be removed

**Comment by Jashpatel1** [comment]
@geraldmaboshe Just one more change, can you reduce the card size of emeritus so as to fit all in one row as per in the current website?

**Comment by geraldmaboshe** [comment]
> @geraldmaboshe Just one more change, can you reduce the card size of emeritus so as to fit all in one row as per in the current website?

Cool, on it!

File: src/sections/Meshmates-grid/Meshmates.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
+import React from "react";
+import { useStaticQuery, graphql } from "gatsby";
+import MeshmatesGrid from "../Meshmates-grid/index";
+
+const Meshmates=props => {
+    const data=useStaticQuery(
```
Comment (Tanuj22) [inline src/sections/Meshmates-grid/Meshmates.js:6]: use this query in the MeshmatesGrid page itself. No need for a separate file

File: src/sections/Meshmates-grid/ProgramDetails.js [hunk 10]
```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+import Program from '../../assets/images/programs/Layer5-mentor-program.png'
+
+export default function ProgramDetails() {
```
Comment (Tanuj22) [inline src/sections/Meshmates-grid/ProgramDetails.js:4]: Please use ES6 syntax (arrow functions)

**Review by Tanuj22 — CHANGES_REQUESTED** [review]
Nicely done! Also, can you decrease the size of Emeritus cards like in the current site? You can override the CSS of the profileCard. Check individual members page for reference on how to do so

File: src/sections/Meshmates-grid/ProgramDetails.js (hunk 10 above)
Comment (geraldmaboshe) [reply src/sections/Meshmates-grid/ProgramDetails.js:4]: Noted

**Comment by geraldmaboshe** [comment]
> @geraldmaboshe Just one more change, can you reduce the card size of emeritus so as to fit all in one row as per in the current website?

@Jashpatel1  do I still maintain using ProfileCard here?

**Comment by Jashpatel1** [comment]
> > @geraldmaboshe Just one more change, can you reduce the card size of emeritus so as to fit all in one row as per in the current website?
> 
> @Jashpatel1 do I still maintain using ProfileCard here?

Yes, as Tanuj stated you can override the CSS.

**Comment by geraldmaboshe** [comment]
> > > @geraldmaboshe Just one more change, can you reduce the card size of emeritus so as to fit all in one row as per in the current website?
> > 
> > 
> > @Jashpatel1 do I still maintain using ProfileCard here?
> 
> Yes, as Tanuj stated you can override the CSS.
How do I achieve overriding? Do I need to create a different file with similar content as ProfileCard and create a style file for it?

**Comment by Jashpatel1** [comment]
> How do I achieve overriding? Do I need to create a different file with similar content as ProfileCard and create a style file for it?

You will have to add a style file for emeritus and overload CSS similar to here https://github.com/layer5io/layer5/blob/layer5-ng/src/sections/Member-single/memberSingle.style.js.

File: src/sections/Meshmates-grid/index.js [hunk 11]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { Link } from "gatsby";
+import { Container, Row, Col } from "../../reusecore/Layout";
+import Button from "../../reusecore/Button";
```
Comment (Tanuj22) [inline src/sections/Meshmates-grid/index.js:4]: @geraldmaboshe can you remove this. Then we are good to go

Comment (geraldmaboshe) [reply src/sections/Meshmates-grid/index.js:4]: > @geraldmaboshe can you remove this. Then we are good to go

`import Button from "../../reusecore/Button";` only right?

Comment (Tanuj22) [reply src/sections/Meshmates-grid/index.js:4]: Yes

Comment (geraldmaboshe) [reply src/sections/Meshmates-grid/index.js]: Done

**Review by Tanuj22 — APPROVED** [review]
Thank you :100:

**Pull request merged**